airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/DebeziumRecordPublisher.java

```diff
@@ -103,10 +108,15 @@ public void close() throws Exception {
         engine.close();
       }
 
-      // announce closure only engine is off.
-      hasClosed.set(true);
+      // wait for closure before shutting down executor service
+      engineLatch.await(5, TimeUnit.MINUTES);
 
+      // shut down and await for thread to actually go down
       executor.shutdown();
+      executor.awaitTermination(5, TimeUnit.MINUTES);
+
+      // after the engine is completely off, we can mark this as closed
+      hasClosed.set(true);
```

Comment:
i think this can go before the execute shutdown. it just had to be after the latch. not that its'a  big difference either way.

Reply:
Yeah I didn't notice any difference. I just wanted to be as conservative as possible.